broadcast/src/buffered/ingress.rs

```diff
@@ -9,103 +10,160 @@ use futures::{
 use std::marker::PhantomData;
 
 /// Message types that can be sent to the `Mailbox`
-pub enum Message<P: Array, D: Digest, M: Digestible<D>> {
+pub enum Message<P: Array, Di: Digest, Dd: Digest, M: Identifiable<Di> + Digestible<Dd>> {
     /// Broadcast a [`Message`](crate::Broadcaster::Message) to the network.
     ///
     /// The responder will be sent a list of peers that received the message.
     Broadcast {
+        recipients: Recipients<P>,
         message: M,
         responder: oneshot::Sender<Vec<P>>,
     },
 
     /// Subscribe to receive a message by digest.
     ///
-    /// The responder will be sent the message when it is available; either instantly (if cached) or
-    /// when it is received from the network. The request can be canceled by dropping the responder.
+    /// The responder will be sent the first message for an identity when it is available; either
+    /// instantly (if cached) or when it is received from the network. The request can be canceled
+    /// by dropping the responder.
     Subscribe {
-        digest: D,
+        sender: Option<P>,
```

Comment:
nit: the use of `sender` can be a bit confusing in this context due to the field called `responder`

consider another name like `peer` (?)

Reply:
In this context, we expect this to be a peer so that's fair. I'll update.